--- GiNaC-1.2.1/ginac/inifcns_nstdsums.cpp.orig	2004-03-18 15:03:24.000000000 +0100
+++ GiNaC-1.2.1/ginac/inifcns_nstdsums.cpp	2004-08-03 20:35:56.041327089 +0200
@@ -218,7 +218,7 @@
 		res = res + (*it) * factor;
 		it++; // should we check it? or rely on initsize? ...
 		i++;
-	} while (res != resbuf);
+	} while ((res != resbuf) && (i <= Xn[0].size()));
 	return res;
 }
 
@@ -255,7 +255,7 @@
 		res = res + (*it) * factor;
 		it++; // should we check it? or rely on initsize? ...
 		i++;
-	} while (res != resbuf);
+	} while ((res != resbuf) && ((i - 1) <= Xn[n-2].size()));
 	return res;
 }
 
